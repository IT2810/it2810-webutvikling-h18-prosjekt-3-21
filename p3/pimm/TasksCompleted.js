import React, { Component } from "react";
import { ScrollView, AsyncStorage } from "react-native";
import { styles } from "./HomeScreen";
import { Task } from "./Task";

export class TasksCompleted extends Component {
  constructor(props) {
    super(props);
    this.state.parent = props.parent;
  }

  state = {
    tasksArray: [],
    parent: null
  };
  _retrieveData = async () => {
    let taskArr = [];
    try {
      AsyncStorage.getAllKeys((err, keys) => {
        AsyncStorage.multiGet(keys, (err, stores) => {
          stores.map((result, i, store) => {
            let task = JSON.parse(store[i][1]);
            if (task !== null) {
              if (task.isCompleted === true) {
                taskArr.push(task);
              }
            }
          });
        }).then(() => {
          this.setState({
            tasksArray: taskArr
          });
        });
      });
    } catch (error) {
      console.log(error);
      alert("Error retrieving data");
    }
  };

  componentDidMount() {
    this._retrieveData();
  }

  updateTasks() {
    this._retrieveData();
  }

  render() {
    return (
      <ScrollView
        contentContainerStyle={styles.taskContainer}
        scrollEnabled={true}
        alwaysBounceVertical={false}
      >
        {this.state.tasksArray.map(task => (
          <Task
            id={task.id_str}
            taskdescription={task.taskDesc}
            key={task.id}
            parent={this}
          />
        ))}
      </ScrollView>
    );
  }
}
