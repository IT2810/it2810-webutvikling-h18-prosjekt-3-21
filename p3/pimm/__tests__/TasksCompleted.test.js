import * as TestRenderer from "react-test-renderer";
import React from "react";
import TasksCompleted from "../TasksCompleted";

beforeAll(() => {
  tree = TestRenderer.create(<TasksCompleted />);
  TasksToDoTest = TestRenderer.create(<TasksCompleted />).getInstance();
});

test("Snapshot test of TasksCompleted.js", () => {
  expect(tree.toJSON).toMatchSnapshot();
});
