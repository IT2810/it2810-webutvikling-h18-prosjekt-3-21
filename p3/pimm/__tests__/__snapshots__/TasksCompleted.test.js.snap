// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of TasksCompleted.js 1`] = `[Function]`;
